### PR TITLE
Feature: new visualization plugin

### PR DESCRIPTION
**Pull request by MatiasArriola**
### :pushpin: References

-   **Issue:** https://app.clickup.com/t/8696vv89u #8696vv89u

### :memo: Implementation

- I initially removed all references to the .js plugin, but then realized that event visualizations wouldn't work with its iframe replacement, so I kept both implementations.
- Now legacy Plugin scripts are loaded only if required
- Created a separate `PluginVisualizationRepository`. Each `get` is performed individually instead of batching all the requests at the repository level.
- `eventCharts` and `eventReports` are unified into `eventVisualizations`, as per DHIS docs.
- I have left a placehoder for OrgUnit filtering.

### :fire: Testing

- Use any DHIS 39+ instance with some dashboards loaded to test.
- Older versions may work, as long as they implement the `eventVisualizations` endpoint.
- During testing, I encountered intermittent issues that I still need to investigate and couldn't reproduce again. These may be related to the DHIS instances used.
  - Inside the Map Visualizations, a form appeared instead of the map.
  - iframes stopped reporting installationStatus correctly
  - Timeout errors in the postRobot communication

### CODE REVIEW

File: src/webapp/hooks/useLegacyVisualizationPlugin.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    const snackbar = useSnackbar();
+    const pluginFileName = useLegacyVisualizationScriptUrl(dashboardItem.legacyReportType);
+    const dhisUrl = useDhis2Url();
+    const [state, setState] = React.useState<LoaderState<undefined>>({ type: "loading" });
+    React.useEffect(() => {
+        loadJsPlugin(pluginFileName)
+            .then(() => {
+                const plugin = window[dashboardItem.legacyReportType];
+                if (!plugin) {
+                    throw new Error(`Legacy plugin "${dashboardItem.legacyReportType}" not found`);
```
Comment (eperedo) [inline src/webapp/hooks/useLegacyVisualizationPlugin.ts:19]: Let's use `i18n.t` for any message displayed in the UI.

**Review by eperedo — CHANGES_REQUESTED** [review]
Thanks @MatiasArriola! It looks good, just a few minor changes in the code.

I was able to load almost every visualization but not this dashboard with eventReport (I understand we should use the legacy plugin for this one, right?)

https://play.im.dhis2.org/stable-2-41-2/dhis-web-dashboard/#/GNlS3sXJZFE

image

image

**Comment by MatiasArriola** [comment]
Thanks @eperedo !

I've implemented the corrections and I will keep them in mind for future changes.

Regarding that particular dashboard, I believe I missed it because it wasn't there past week, and also most of my testing was done against the 2-40-6 instance.
The Item is not loading because when we make the metadata GET request to build the dashboard, the `eventVisualization` field is not included in the response in that  dashboardItem. As a result, the item is filtered out.
Here is a working example of a dashboard that uses event charts and event reports: https://play.im.dhis2.org/stable-2-41-2/dhis-web-dashboard/#/eZSIccgeq94. For these dashboardItems, the `eventVisualization` field is returned. 

I don't understand why that dashboard doesn't return the `eventVisualization` field in the item, while others do. 
I've stopped using the `eventReports` entity to align with the [docs recommendations](https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-241/visualizations.html#webapi_event_visualization). 

If needed for legacy support, we could add back the `eventChart` and `eventReport` to the metadata GET fields requested.

**Review by eperedo — APPROVED** [review]
Thanks @MatiasArriola, looks good to me. 

You're right about the `eventVisualization` I've tested in the test environment of the client that is working with the app and visualizations include the eventVisualization object. So let's keep your implementation.

**Pull request merged**